packages/mui-material/src/Container/Container.js

```diff
@@ -1,113 +1,26 @@
-import * as React from 'react';
+/* eslint-disable material-ui/mui-name-matches-component-name */
 import PropTypes from 'prop-types';
-import clsx from 'clsx';
-import { unstable_composeClasses as composeClasses } from '@mui/base';
-import useThemeProps from '../styles/useThemeProps';
-import styled from '../styles/styled';
-import { getContainerUtilityClass } from './containerClasses';
+import { createContainer } from '@mui/system';
 import capitalize from '../utils/capitalize';
+import styled from '../styles/styled';
+import useThemeProps from '../styles/useThemeProps';
 
-const useUtilityClasses = (ownerState) => {
-  const { classes, fixed, disableGutters, maxWidth } = ownerState;
-
-  const slots = {
-    root: [
-      'root',
-      maxWidth && `maxWidth${capitalize(String(maxWidth))}`,
-      fixed && 'fixed',
-      disableGutters && 'disableGutters',
-    ],
-  };
-
-  return composeClasses(slots, getContainerUtilityClass, classes);
-};
-
-const ContainerRoot = styled('div', {
-  name: 'MuiContainer',
-  slot: 'Root',
-  overridesResolver: (props, styles) => {
-    const { ownerState } = props;
+const Container = createContainer({
+  createStyledComponent: styled('div', {
+    name: 'MuiContainer',
+    slot: 'Root',
+    overridesResolver: (props, styles) => {
+      const { ownerState } = props;
 
-    return [
-      styles.root,
-      styles[`maxWidth${capitalize(String(ownerState.maxWidth))}`],
-      ownerState.fixed && styles.fixed,
-      ownerState.disableGutters && styles.disableGutters,
-    ];
-  },
-})(
-  ({ theme, ownerState }) => ({
-    width: '100%',
-    marginLeft: 'auto',
-    boxSizing: 'border-box',
-    marginRight: 'auto',
-    display: 'block', // Fix IE11 layout when used with main.
-    ...(!ownerState.disableGutters && {
-      paddingLeft: theme.spacing(2),
-      paddingRight: theme.spacing(2),
-      [theme.breakpoints.up('sm')]: {
-        paddingLeft: theme.spacing(3),
-        paddingRight: theme.spacing(3),
-      },
-    }),
+      return [
+        styles.root,
+        styles[`maxWidth${capitalize(String(ownerState.maxWidth))}`],
+        ownerState.fixed && styles.fixed,
+        ownerState.disableGutters && styles.disableGutters,
+      ];
+    },
```

Comment:
Is this logic only here temporarily until v6? Should we have a TODO so we don't forget to remove it at some point?

Reply:
The point here is that for Joy and Material UI the overrides resolver are different. In Joy, we do not have style overrides with keys, it is a leftover from Material UI. I will add a comment for updating this in v6.